doc/docs.md

```diff
@@ -1905,6 +1905,61 @@ fn announce(s Speaker) {
 ```
 For more information, see [Dynamic casts](#dynamic-casts).
 
+```v
+struct Cat {}
+
+interface Animal {}
+
+fn (a Animal) is_animal() {
```

Comment:
Can we have a more realistic example of where this feature is useful?

Reply:
The first thing that came to my mind was using an interface with methods to model adoption by wrapping the Animal interface with an Adoptable interface. The Adoptable has an adopt() method that returns an Animal, which seems to me like a reasonable functionally styled way to prevent animals from being adopted twice (as opposed to mutating an adoptable boolean field in Animal).
https://pastebin.com/raw/rWbrWJdU

However, this generates a C error, so I'm gonna have to think up a new idea.

Reply:
```v
struct Cat {}

fn (c Cat) speak() string {
		return 'meow!'
}

interface Adoptable {}

fn (a Adoptable) adopt() ?&Cat {
		if a is Cat {
				return a
		}	else {
				return error('This cannot be adopted.')
		}
}

fn new_adoptable() Adoptable {
		return Cat{}
}

fn main() {
		adoptable := new_adoptable()
		cat := adoptable.adopt() or { return }
		println(cat.speak())
}
```
A simpler example which wraps Cat in this way, rather than Animal, seems to work. Is this usable?